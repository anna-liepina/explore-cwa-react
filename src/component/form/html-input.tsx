import React from 'react';
import type { InputHTMLAttributes } from 'react';
import composeInput from './compose-form-field';

const HTMLInput: React.FC<InputHTMLAttributes<HTMLInputElement>> = (props) => <input {...props} />;

export default composeInput(HTMLInput);
export { HTMLInput as PureHTMLInput };
