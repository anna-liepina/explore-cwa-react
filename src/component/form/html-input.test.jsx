import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { PureHTMLInput as HTMLInput } from './html-input';

describe('<HTMLInput/>', () => {
    const props = {
    };

    const optionalProps = {
        'data-cy': 'optProps.data-cy',
        className: 'optProps.className',
        value: 'optProps.value',
        onChange: jest.fn(),
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment } = render(<HTMLInput {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props', () => {
            const { asFragment } = render(<HTMLInput {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
