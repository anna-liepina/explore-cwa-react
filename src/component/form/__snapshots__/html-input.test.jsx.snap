// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HTMLInput/> render with default/required props 1`] = `
<DocumentFragment>
  <input />
</DocumentFragment>
`;

exports[`<HTMLInput/> render with optional/required props 1`] = `
<DocumentFragment>
  <input
    class="optProps.className"
    data-cy="optProps.data-cy"
    value="optProps.value"
  />
</DocumentFragment>
`;
